python-checks/src/main/java/org/sonar/python/checks/UnusedFunctionParametersCheck.java

```diff
@@ -0,0 +1,88 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import javax.annotation.CheckForNull;
+import org.sonar.check.Rule;
+import org.sonar.plugins.python.api.PythonSubscriptionCheck;
+import org.sonar.plugins.python.api.SubscriptionContext;
+import org.sonar.plugins.python.api.symbols.Symbol;
+import org.sonar.plugins.python.api.symbols.Usage;
+import org.sonar.plugins.python.api.tree.ClassDef;
+import org.sonar.plugins.python.api.tree.ExpressionStatement;
+import org.sonar.plugins.python.api.tree.FunctionDef;
+import org.sonar.plugins.python.api.tree.Parameter;
+import org.sonar.plugins.python.api.tree.Tree.Kind;
+import org.sonar.python.semantic.SymbolUtils;
+
+@Rule(key = "S1172")
+public class UnusedFunctionParametersCheck extends PythonSubscriptionCheck {
+
+  private static final String MESSAGE = "Remove the unused function parameter \"%s\".";
+
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(Kind.FUNCDEF, ctx -> checkFunctionParameter(ctx, ((FunctionDef) ctx.syntaxNode()), ((FunctionDef) ctx.syntaxNode()).localVariables()));
+  }
+
+  private static void checkFunctionParameter(SubscriptionContext ctx, FunctionDef functionDef, Set<Symbol> symbols) {
+    if (CheckUtils.isCallingLocalsFunction(functionDef) || maybeOverridingMethod(functionDef) || isInterfaceMethod(functionDef)) {
+      return;
+    }
+
+    symbols.stream()
+      .map(UnusedFunctionParametersCheck::getUnusedParameter)
+      .filter(Objects::nonNull)
+      .forEach(param -> ctx.addIssue(param, String.format(MESSAGE, param.name().name())));
+  }
+
+  @CheckForNull
+  private static Parameter getUnusedParameter(Symbol symbol) {
+    if ("self".equals(symbol.name())) {
+      return null;
+    }
+    List<Usage> usages = symbol.usages();
+    if (usages.size() == 1 && usages.get(0).tree().parent().is(Kind.PARAMETER)) {
+      return (Parameter) usages.get(0).tree().parent();
+    }
+    return null;
+  }
+
+  private static boolean isInterfaceMethod(FunctionDef functionDef) {
+    return functionDef.body().statements().stream()
+      .allMatch(statement -> statement.is(Kind.PASS_STMT, Kind.RAISE_STMT)
+        || (statement.is(Kind.EXPRESSION_STMT) && isStringExpression((ExpressionStatement) statement)));
+  }
+
+  private static boolean isStringExpression(ExpressionStatement stmt) {
+    return stmt.expressions().stream().allMatch(expr -> expr.is(Kind.STRING_LITERAL));
+  }
+
+  private static boolean maybeOverridingMethod(FunctionDef functionDef) {
```

Comment:
That's really the kind of feature I would expect from the symbol table. Let's keep it like that for the moment.